experiment/scheduler.py

```diff
@@ -130,7 +130,7 @@ def end_expired_trials(experiment_config: dict):
     if not expired_instances:
         return
 
-    if not delete_instances(expired_instances, experiment_config):
+    if not experiment_utils.is_local_experiment() and not delete_instances(expired_instances, experiment_config):
```

Comment:
line is almost certainly too long.
I think nesting here is fine. Or do `if local: bulksave; return;`